### PR TITLE
M3DB user and Namespace + Database + capabilities data

### PR DESCRIPTION
**Pull request by lpatte**
- M3db user (resource/datasource/import/doc) 
- M3db namespace (resource/datasource/import/doc) 
- database (resource/datasource/import/doc) 
- capabilities (datasource/doc)
- Add field `kafka_rest_api` and `opensearch_acls_enabled` to `cloud_project_database` (fix https://github.com/ovh/terraform-provider-ovh/issues/309)
- Fix Mongodb user output by adding a `StateFunc` to name attribut

### CODE REVIEW

File: website/docs/r/cloud_project_database_m3db_namespace.html.markdown [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+layout: "ovh"
+page_title: "OVH: cloud_project_database_m3db_namespace"
+sidebar_current: "docs-ovh-resource-cloud-project-database-m3db-namespace"
+description: |-
+  Creates a namespace for a m3db cluster associated with a public cloud project.
+---
+
+# ovh_cloud_project_database_m3db_namespace
+
+Creates a namespace for a m3db cluster associated with a public cloud project.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_namespace.html.markdown:11]:
```suggestion
Creates a namespace for a M3DB cluster associated with a public cloud project.
```

File: website/docs/r/cloud_project_database_m3db_user.html.markdown [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+layout: "ovh"
+page_title: "OVH: cloud_project_database_m3db_user"
+sidebar_current: "docs-ovh-resource-cloud-project-database-m3db-user"
+description: |-
+  Creates an user for a m3db cluster associated with a public cloud project.
+---
+
+# ovh_cloud_project_database_m3db_user
+
+Creates an user for a m3db cluster associated with a public cloud project.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_user.html.markdown:11]:
```suggestion
Creates a user for a M3DB cluster associated with a public cloud project.
```

File: website/docs/r/cloud_project_database_m3db_user.html.markdown [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+}
+```
+* `create` - (Default 20m)
+* `update` - (Default 20m)
+* `delete` - (Default 20m)
+
+## Import
+
+OVHcloud Managed m3db clusters users can be imported using the `service_name`, `cluster_id` and `id` of the user, separated by "/" E.g.,
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_user.html.markdown:75]:
```suggestion
OVHcloud Managed M3DB clusters users can be imported using the `service_name`, `cluster_id` and `id` of the user, separated by "/" E.g.,
```

File: website/docs/d/cloud_project_database_capabilities.html.markdown [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  * `min_node_number` - Minimum nodes of the cluster.
+  * `network` - Type of network.
+  * `plan` - Plan name.
+  * `region` - Region name.
+  * `start_date` - Date of the release of the product.
+  * `status` - Status of the availability.
+  * `step_disk_size` - Flex disk size step in GB.
+  * `upstream_end_of_life` - End of life of the upstream product.
+  * `version` - Version name.
+* `engine` - Database engines available.
```
Comment (scraly) [inline website/docs/d/cloud_project_database_capabilities.html.markdown:55]:
```suggestion
* `engines` - Database engines available.
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: website/docs/r/cloud_project_database_m3db_namespace.html.markdown [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+data "ovh_cloud_project_database" "m3db" {
+  service_name  = "XXX"
+  engine        = "m3db"
+  id            = "ZZZ"
+}
+
+resource "ovh_cloud_project_database_m3db_namespace" "namespace" {
+  service_name  = data.ovh_cloud_project_database.m3db.service_name
+  cluster_id    = data.ovh_cloud_project_database.m3db.id
+  group         = "mygroup"
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_namespace.html.markdown:25]: `retention_period_duration` instead? With an example of the required format please :)

File: website/docs/r/cloud_project_database_m3db_namespace.html.markdown [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  name          = "johndoe"
```
Comment (scraly) [inline website/docs/r/cloud_project_database_m3db_namespace.html.markdown:26]:
```suggestion
  name          = "my-namespace"
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: website/docs/d/cloud_project_database_m3db_namespaces.html.markdown [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+* `cluster_id` - (Required) Cluster ID
+
+## Attributes Reference
+
+`id` is set to the md5 sum of the list of all namespaces ids. In addition,
+the following attributes are exported:
+
+* `cluster_id` - See Argument Reference above.
+* `service_name` - See Argument Reference above.
+* `namespace_ids` - The list of namespaces ids of the m3db cluster associated with the project.
```
Comment (scraly) [inline website/docs/d/cloud_project_database_m3db_namespaces.html.markdown:40]:
```suggestion
* `namespace_ids` - The list of namespaces ids of the M3DB cluster associated with the project.
```

File: website/docs/index.html.markdown [hunk 8]
```diff
@@ -127,6 +127,8 @@ variables must also be set:
 
 * `OVH_CLOUD_PROJECT_DATABASE_KAFKA_VERSION_TEST` - The version of the kafka to test. if not set `OVH_CLOUD_PROJECT_DATABASE_VERSION_TEST` is use.
 
+* `OVH_CLOUD_PROJECT_DATABASE_M3DB_VERSION_TEST` - The version of the mongodb to test. if not set `OVH_CLOUD_PROJECT_DATABASE_VERSION_TEST` is use.
```
Comment (scraly) [inline website/docs/index.html.markdown:130]:
```suggestion
* `OVH_CLOUD_PROJECT_DATABASE_M3DB_VERSION_TEST` - The version of the M3DB to test. if not set `OVH_CLOUD_PROJECT_DATABASE_VERSION_TEST` is use.
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: ovh/data_cloud_project_database_m3db_namespace.go [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+				Computed:    true,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Description: "Type of namespace",
+				Computed:    true,
+			},
+			"writes_to_commit_log_enabled": {
+				Type:        schema.TypeBool,
+				Description: "Defines whether M3db will include writes to this namespace in the commit log",
```
Comment (scraly) [inline ovh/data_cloud_project_database_m3db_namespace.go:74]:
```suggestion
				Description: "Defines whether M3DB will include writes to this namespace in the commit log",
```

File: website/docs/r/cloud_project_database.html.markdown [hunk 10]
```diff
@@ -182,13 +196,17 @@ The following arguments are supported:
...
+* `kafka_rest_api` -  (Optional) Defines whether the REST API is enabled on a kafka cluster
+
 * `nodes` - (Required, Minimum Items: 1) List of nodes object.
   Multi region cluster are not yet available, all node should be identical.
   * `network_id` - (Optional, Forces new resource) Private network id in which the node should be deployed.
   * `region` - (Required, Forces new resource) Public cloud region in which the node should be deployed.
     Ex: "GRA'.
   * `subnet_id` - (Optional, Forces new resource) Private subnet ID in which the node is.
 
+* `opensearch_acls_enabled` -  (Optional) Defines whether the acls are enabled on an Opensearch cluster
```
Comment (scraly) [inline website/docs/r/cloud_project_database.html.markdown:208]:
```suggestion
* `opensearch_acls_enabled` -  (Optional) Defines whether the ACLs are enabled on an OpenSearch cluster
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by albundy83** [comment]
Big merge :)
Is there the support to create databases with PostgreSQL ?

**Pull request merged**